### ISSUE TITLE
Performance/Casecmp not equivalent to string.downcase==string

### ISSUE THREAD

**Issue by aarondcohen** [bbatsov/rubocop#2893]
Although, `casecmp` is a faster way to do case-insensitive string comparison than applying a case function, not all cased-comparisons are case insensitive.  Take the example `string.downcase==string`.  We are actually asking: is the string lowercase?  There are alternatives like regexes, however they are far less clear in code and can have annoying corner cases.

**Comment by segiddins** [comment]
This ought to be fixed on master now

**Comment by aarondcohen** [comment]
Cool.  It might be worth adding the following tests.  Otherwise, I look forward to using the next version of master.

```ruby
    it "doesn't report an offense for str.#{selector} == str" do
      inspect_source(cop, "str.#{selector} == str")
      expect(cop.offenses).to be_empty
    end

    it "doesn't report an offense for str == str.#{selector}" do
      inspect_source(cop, "str == str.#{selector}")
      expect(cop.offenses).to be_empty
    end
```

**Comment by alexdowad** [comment]
If tests like you show are not there, please open a PR. I can't remember if
I already added them or not.

**Comment by alexdowad** [comment]
Equivalent tests are already there. This can be closed.
